### ISSUE TITLE
Movement speed coupled to camera orientation

### ISSUE THREAD

**Issue by maystey** [GDQuest/godot-3d-mannequin#39]
**I'm submitting a...**

- [x ] Bug report.
- [ ] Feature request.

## Bug report ##

**What is the current behavior?**
Movement speed is linked to the camera angle. In particular if the camera angle is pointing upwards, the speed is greatly reduced.

**What is the expected behavior?**
I assume that the speed is meant to be independent of the camera angle.

**Tell us the steps to reproduce the bug, and if possible share a minimal demo of the problem.** 
1. Enter the main 'Game' scene
2. Point the camera into the sky.
3. Run around.

## Feature request ##

**Describe the problem you're trying to solve.**

**Tell us which solutions you've explored, the solution you would pick, and why you think it would be the best for everyone.**

## Other ##

**Other information** 
I have a PR waiting in the wings if this is indeed a bug.

**Comment by NathanLovato** [comment]
I'm not sure if it should be considered a bug. The character moves in the camera's direction when pressing forward, so if you aim it at the ground it's normal that it moves slower than normal along the ground. But I see how it could cause issues with slopes.

What matters most to me is that the character is reactive and feels somewhat good in the hand. If removing that improves the movement's feel or the responsivity of the character, then please go ahead and open a PR.

**Comment by NathanLovato** [comment]
And thanks for taking the time to open issues and contribute, it's much appreciated!

**Comment by maystey** [comment]
You're welcome. You've given us excellent resources, it's the least I can do.

As far as the feel goes, I'm not sure. I wasn't entirely sure if this was a bug in the first place. I found it surprising that the character slowed down when looking up, but it doesn't necessarily feel bad. (Maybe with some look IK it will even seem like the most natural choice - just a thought.)

I'll push the branch where I've implemented the changes to my fork and link it to this issue thread. At the very least people can decide for themselves what they prefer and use the changes in their own implementations if they want.

**Comment by Braboware** [comment]
I agree that the feel is better when move speed isn't linked to the angle of the camera. I was about to open an issue myself but I saw that this was open.
